flow/data_pipeline/query.py

```diff
@@ -88,27 +103,390 @@ class QueryStrings(Enum):
+                AVG(speed) AS speed_avg,
+                AVG(speed) + STDDEV(speed) AS speed_upper_bound,
+                AVG(speed) - STDDEV(speed) AS speed_lower_bound,
+                AVG(target_accel_no_noise_with_failsafe) AS accel_avg,
+                AVG(target_accel_no_noise_with_failsafe) + STDDEV(target_accel_no_noise_with_failsafe) AS accel_upper_bound,
+                AVG(target_accel_no_noise_with_failsafe) - STDDEV(target_accel_no_noise_with_failsafe) AS accel_lower_bound,
+                AVG(energy_joules) AS cumulative_energy_avg,
+                AVG(energy_joules) + STDDEV(energy_joules) AS cumulative_energy_upper_bound,
+                AVG(energy_joules) - STDDEV(energy_joules) AS cumulative_energy_lower_bound
+            FROM cumulative_energy
+            GROUP BY 1, 2
+            HAVING 1 = 1
+                AND COUNT(DISTINCT time_step) > 1
+        ), binned_energy_start_end AS (
+            SELECT DISTINCT
+                source_id,
+                id,
+                CAST(time_step/60 AS INTEGER) * 60 AS time_seconds_bin,
+                FIRST_VALUE(energy_joules)
+                    OVER (PARTITION BY id, CAST(time_step/60 AS INTEGER) * 60
+                    ORDER BY time_step ASC) AS energy_start,
+                LAST_VALUE(energy_joules)
+                    OVER (PARTITION BY id, CAST(time_step/60 AS INTEGER) * 60
+                    ORDER BY time_step ASC) AS energy_end
+            FROM cumulative_energy
+        ), binned_energy AS (
+            SELECT
+                source_id,
+                time_seconds_bin,
+                AVG(energy_end - energy_start) AS instantaneous_energy_avg,
+                AVG(energy_end - energy_start) + STDDEV(energy_end - energy_start) AS instantaneous_energy_upper_bound,
+                AVG(energy_end - energy_start) - STDDEV(energy_end - energy_start) AS instantaneous_energy_lower_bound
+            FROM binned_energy_start_end
+            GROUP BY 1, 2
+        )
+        SELECT
+            COALESCE(bce.source_id, be.source_id) AS source_id,
+            COALESCE(bce.time_seconds_bin, be.time_seconds_bin) AS time_seconds_bin,
+            bce.cumulative_energy_avg,
+            bce.cumulative_energy_lower_bound,
+            bce.cumulative_energy_upper_bound,
+            bce.speed_avg,
+            bce.speed_upper_bound,
+            bce.speed_lower_bound,
+            bce.accel_avg,
+            bce.accel_upper_bound,
+            bce.accel_lower_bound,
+            be.instantaneous_energy_avg,
+            be.instantaneous_energy_upper_bound,
+            be.instantaneous_energy_lower_bound
+        FROM binned_cumulative_energy bce 
+        FULL OUTER JOIN binned_energy be ON 1 = 1
+            AND bce.source_id = be.source_id
+            AND bce.time_seconds_bin = be.time_seconds_bin
+        ORDER BY time_seconds_bin ASC
+        ;"""
+
+    LEADERBOARD_CHART_AGG = """
+        SELECT
+            date AS submission_date,
```

Comment:
is `date` the date of submission or query? i want date of submission, but i think it's date of query.